build.gradle

```diff
@@ -794,39 +795,49 @@ ext.makeWinTask = { args ->
                 def exe = exePresent('iscc') ?  'iscc' : file('ci/iscc')
                 commandLine exe, "${destinationDir}/OmegaT.iss"
             }
-            ant.move file: "${destinationDir}/${installerBasename}.exe",
-                todir: distsDir
+            project.copy {
+                from("${destinationDir}/${installerBasename}.exe")
+                into(distsDir)
+            }
         }
     }
     def signedTaskName = "${args.name}Signed"
-    task(signedTaskName) {
+    def signedExe = "${distsDir}/${installerBasename}_Signed.exe"
+    task(type: Exec, signedTaskName) {
         onlyIf {
             // Set these in e.g. local.properties
-            def props = ['winCodesignFile', 'winCodesignPassword', 'winCodesignTimestampUrl']
+            def props = ['winCodesignCert', 'winCodesignPassword']
             conditions([props.every { project.hasProperty(it) }, 'Code signing properties not set'],
                        [file(installerExe).file, 'Unsigned installer not built'],
-                       [exePresent('osslsigncode') || exePresent('docker'),
-                        'osslsigncode or Docker not installed'])
+                       [exePresent('osslsigncode'), 'osslsigncode not installed'])
         }
-        def signedExe = "${distsDir}/${installerBasename}_Signed.exe"
+        def pkcs11module = project.hasProperty('pkcs11module')? project.property('pkcs11module') :
+                '/usr/lib/x86_64-linux-gnu/pkcs11/opensc-pkcs11.so'
+        def pkcs11time = project.hasProperty('winCodesignTimestampUrl')? project.property('winCodesignTimestampUrl') :
+                'http://time.certum.pl/'
+        def pkcs11cert = project.hasProperty('winCodesignCert')? project.property('winCodesignCert') : ''
+        def pkcs11pass = project.hasProperty('winCodesignPassword')? project.property('winCodesignPassword') : ''
```

Comment:
If `winCodesignPassword` not provided, we can use `-askpass` option when it does not run on headless environment, other than CI. we can check project property 'IsCIEnv' .

Reply:
It is better to avoid interaction with developer because of stability of gradle script.